cirq/linalg/diagonalize_test.py

```diff
@@ -68,53 +68,38 @@ def random_bi_diagonalizable_pair(
     b = cirq.dot(u, z, v)
     return a, b
 
+def _get_assert_diagonalized_by_str(m, p, d):
+    # coverage: ignore
+    return 'm.round(3) : %f, p.round(3) : %f, ' \
+        'np.log10(np.abs(p.T @ m @ p)).round(2): %f' %(np.round(m, 3),
+                                                       np.round(p, 3),
+                                                       np.log10(np.abs(d))\
+                                                       .round(2))
 
 def assert_diagonalized_by(m, p, atol: float = 1e-8):
     d = p.T.dot(m).dot(p)
 
-    try:
-        assert cirq.is_orthogonal(p)
-        assert cirq.is_diagonal(d, atol=atol)
-    except AssertionError:
-        # coverage: ignore
-
-        print("m.round(3)")
-        print(np.round(m, 3))
-
-        print("p.round(3)")
-        print(np.round(p, 3))
-
-        print("np.log10(np.abs(p.T @ m @ p)).round(2)")
-        print(np.log10(np.abs(d)).round(2))
-
-        raise
+    # coverage: ignore
```

Comment:
Why is this ignoring coverage?

Reply:
` _get_assert_diagonalized_by_str(m, p, d)` is not called if the `assert` passes, so `coverage` never picks this line in that case.

Reply:
There should be a test that triggers the failure case. If there isn't, we should add one.

```python
with pytest.raise(AssertError, match="bla bla"):
    ...
```

Reply:
No entirely sure about this. Lemme remove and see.

Reply:
It's possible that the old version didn't have one because the `print` lines would end up in the output. This is part of the benefit of switching to the `assert bool, str` style: we can properly cover.

Reply:
Seems like my local `pytest-and-incremental-coverage` doesn't reflect remote. Will add tests that trigger failure cases tomorrow.